### ISSUE TITLE
Selecting text via shift+directional key not working as expected (arrow-, home-, end-key)

### ISSUE THREAD

**Issue by fallmake** [zeit/hyper#1642]
1. Write any text in shell
2. using shift+home key does nothing (expected: select text from current position to home position)
3. using shift+end key does nothing (expected: select text from current position to end of line)
4. using shift+left arrow marks text visually, but copy doesn't work with a selection made that way.

Selecting text via mouse looks different (selection color differs) and works.

Encountered with powershell.exe
-

Hyper a35f39f
Electron 1.4.15
win32 x64 10.0.14393

**Comment by MaxFrost** [comment]
To extend this, things like ctrl+shift+arrow don't work as they do elsewhere . Ctrl+Arrow is "move to end of current word", and if you hold shift, it should also select the word as it moves. This behavior is the same regardless if you're in Word, Commandprompt, or Powershell. I believe the same shortcuts are also available in Linux and OSX, with OSX using command instead of ctrl.

Hyper v. 1.3.1
Win 10 x64

**Comment by MatthiasWinkelmann** [comment]
Shells on Unix traditionally don't have a concept of 'selection', so I believe this is working as intended. It spans across domains of both shell (cursor position) and terminal application (selection) and is probably quite difficult to get right across OS and shells.

**Comment by kleinfreund** [comment]
The Ubuntu Terminal app does understand these keybindings quite well while Hyper doesn’t.

Roughly documenting the expected features:

- <kbd>Ctrl</kbd><kbd>←</kbd>: Move caret before current word boundary (this is afaik the index of the first letter of the word across most operating systems)
- <kbd>Ctrl</kbd><kbd>→</kbd>: Move caret after current word boundary (the behavior on Windows being that the the caret position will be the index of the first letter of the following word while on Unix systems it’s the index of the following whitespace character)
- <kbd>Ctrl</kbd><kbd>←</kbd>: Move caret to the previous character
- <kbd>Ctrl</kbd><kbd>→</kbd>: Move caret to the next character

**Comment by emiljohnsen** [comment]
This is an issue on Windows, or, well, I think it is. 

<kbd>Ctrl</kbd>+<kbd>Directional/Home/End</kbd> for moving the cursor, 
<kbd>Shift</kbd>+<kbd>Directional/Home/End</kbd> for selecting text. 

Could also be that this is better suited for a plugin? The cmd shell in windows has limited support for these as well, natively, but are sorely missed by me at least. 

What caught me a bit off-guard is that this is the standard behaviour in editable HTML components (`contenteditable`, `input`, `textarea` etc.).

**Comment by simoncorry** [comment]
This still appears to be an issue on Windows.

**Comment by AdsonCicilioti** [comment]
> This still appears to be an issue on Windows.

No. This behavior is present on Linux also. I have this exactly behavior on Arch Linux in multiple Terminal emulators.

**Comment by davidgilbertson** [comment]
This is an issue in macOS too.

I can click+drag with the mouse to select text, why can I not arrow key + alt/cmd to do the same?

Anyone know any good alternatives that allow normal mouse-free text selection?
